plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.tsx

```diff
@@ -160,9 +160,9 @@ export const RepoUrlPicker = (
       {hostType === 'github' && (
         <GithubRepoPicker
           allowedOwners={allowedOwners}
+          onChange={prevState => ({ ...prevState, allowedOwners })}
           rawErrors={rawErrors}
           state={state}
-          onChange={updateLocalState}
```

Comment:
I wonder if there's something inside this handler that's causing the issue, because I thought under the hood it basically does something like `onChange={prevState => ({ ...prevState, allowedOwners })}`

Reply:
Hey, yeah thats what I am looking at indeed now, my initial approach I realize now was not the best. :sweat_smile:

Reply:
Hey @benjdlambert this seems to work:

```  
const updateLocalState = useCallback(
    (owner: RepoUrlPickerState) => {
      setState(prevState => ({ ...prevState, ...owner }));
    },
    [setState],
  );
```

But then the tests fail with this which I am not 100% sure why:

```
  ● RepoUrlPicker › happy path rendering › should render the repo url picker with minimal props

    expect(jest.fn()).toHaveBeenCalledWith(...expected)

    Expected: ObjectContaining {"formData": "github.com?owner=backstage&repo=repo123"}, Anything

    Number of calls: 0

      81 |       fireEvent.click(submitButton);
      82 |
    > 83 |       expect(onSubmit).toHaveBeenCalledWith(
         |                        ^
      84 |         expect.objectContaining({
      85 |           formData: 'github.com?owner=backstage&repo=repo123',
      86 |         }),

      at Object.toHaveBeenCalledWith (components/fields/RepoUrlPicker/RepoUrlPicker.test.tsx:83:24)

  ● RepoUrlPicker › requestUserCredentials › should call the scmAuthApi with the correct params

    expect(jest.fn()).toHaveBeenCalledWith(...expected)

    Expected: {"additionalScope": {"customScopes": {"github": ["workflow"]}, "repoWrite": true}, "url": "https://github.com/backstage/repo123"}

    Number of calls: 0

      160 |       });
      161 |
    > 162 |       expect(mockScmAuthApi.getCredentials).toHaveBeenCalledWith({
          |                                             ^
      163 |         url: 'https://github.com/backstage/repo123',
      164 |         additionalScope: {
      165 |           repoWrite: true,
```

Would you have any idea why? Not sure if after making the change the tests should just work or need to change the test themselves. But in `yarn dev` this change seems to work.

Reply:
Found the issue, a strange one indeed.

Doing this change basically disables the `Owner` input on the second step of creating one of the example templates it won't accept any input :thinking: 

Will continue to debug this.